src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -933,9 +1541,740 @@ bool W3DDisplay::Is_LetterBox_Fading()
+
+        double fps = frameCount / frameTime;
+        double draw_calls = 0;
+        // draw_calls = (double)Debug_Statistics::Get_Draw_Calls();
+        double sorting_polygons = 0;
+        // sorting_polygons = (double)Debug_Statistics::Get_Sorting_Polygons();
+        double dx8_skin_renders = 0;
+        // dx8_skin_renders = (double)Debug_Statistics::Get_DX8_Skin_Renders();
+        fps = std::max(0.1, fps);
+        double fps2 = 1000.0 / fps;
+#ifdef GAME_DEBUG_STRUCTS
+        double fps3 = (double)(PerformanceCounter.QuadPart - m_performanceCounter);
+        fps = std::max(0.1, fps);
+        int frame = g_theGameLogic->Get_Frame() - 15;
+        double fps4;
+
+        if (frame <= 0 || fps3 <= 0.0) {
+            fps4 = 0.0f;
+        } else {
+            fps4 = (double)frame / fps3;
+        }
+
+        double dx8_skin_polygons = 0;
+        // dx8_skin_polygons = (double)Debug_Statistics::Get_DX8_Skin_Polygons();
+
+        if (g_theWriteableGlobalData->m_useFPSLimit) {
+            str1.Format(L"%.2f/%d FPS, ", fps, g_theGameEngine->Get_FPS_Limit());
+        } else {
+            str1.Format(L"%.2f FPS, ", fps);
+        }
+
+        str2.Format(L"%.2fms [cumuFPS=%.2f] draws: %d skins: %d sortP: %d skinP: %d LOD %d",
+            fps2,
+            fps4,
+            (unsigned int)draw_calls,
+            (unsigned int)dx8_skin_renders,
+            (unsigned int)sorting_polygons,
+            (unsigned int)dx8_skin_polygons,
+            g_theWriteableGlobalData->m_terrainLOD);
+        str1.Concat(str3);
+#else
+        str1.Format(L"FPS: %.2f, %.2fms draws: %.2f skins: %.2f sort %.2f",
+            fps,
+            fps2,
+            draw_calls,
+            dx8_skin_renders,
+            sorting_polygons);
+
+        if (g_theWriteableGlobalData->m_useFPSLimit) {
+            str2.Format(L", FPSLock %d", g_theWriteableGlobalData->m_framesPerSecondLimit);
+        }
+
+        str1.Concat(str2);
+#endif
+
+        str3.Format(L"FPS: %.2f", fps);
+        m_benchmarkDisplayString[0]->Set_Text(str3);
+
+        int dx8_polygons = 0;
+        // dx8_polygons = Debug_Statistics::Get_DX8_Polygons();
+        // obsolete stuff checking debug runtime registry key here
+        m_displayStrings[0]->Set_Text(str1);
+
+        str1.Format(L"Frame: %d", g_theGameLogic->Get_Frame());
+        m_displayStrings[1]->Set_Text(str1);
+
+        str1.Format(L"Polygons: per frame %d, per second %d", dx8_polygons, (unsigned int)((double)dx8_polygons * fps));
+        m_displayStrings[2]->Set_Text(str1);
+
+        int dx8_vertices = 0;
+        // dx8_vertices = Debug_Statistics::Get_DX8_Vertices();
+        str1.Format(L"Vertices: %d", dx8_vertices);
+        m_displayStrings[3]->Set_Text(str1);
+
+        int record_texture_size = 0;
+        // record_texture_size = Debug_Statistics::Get_Record_Texture_Size();
+        str1.Format(L"Video RAM: %d", record_texture_size - 0x150000);
```

Comment:
I see all code with `L""` strings is guarded by `#ifdef PLATFORM_WINDOWS` or `#ifdef BUILD_WITH_D3D8`. So L"" is ok. But typically we use `U_CHAR("")`. Leave as is or change?

Reply:
Keep as-is.